jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/AbsoluteUnixPath.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.filesystem;
+
+import com.google.common.base.Preconditions;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.StringJoiner;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Represents a Unix-style path in absolute form (containing all path components relative to the
+ * file system root {@code /}).
+ *
+ * <p>This class is immutable and thread-safe.
+ */
+@Immutable
+public class AbsoluteUnixPath {
+
+  /**
+   * Gets a new {@link AbsoluteUnixPath} from a {@link Path}. The {@code path} must be absolute
+   * (indicated by a non-null {@link Path#getRoot}).
+   *
+   * @param path the absolute {@link Path} to convert to an {@link AbsoluteUnixPath}.
+   * @return a new {@link AbsoluteUnixPath}
+   */
+  public static AbsoluteUnixPath fromPath(Path path) {
+    Preconditions.checkArgument(
+        path.getRoot() != null, "Cannot create AbsoluteUnixPath from non-absolute Path: " + path);
+
+    StringJoiner pathJoiner = new StringJoiner("/", "/", "");
+    for (Path pathComponent : path) {
```

Comment:
I think there's a way to use streams here if you want:
```java
Streams.stream(path.iterator()).collect(Collectors.joining("/","/",""));
```

Reply:
I think since this might be a class that is called many times (for processing many `LayerEntry`s or serializing for cache metadata), we should prefer to use iterative style over streams since streams add a significant amount of overhead.